pubgrep/pubgrep.py

```diff
@@ -0,0 +1,520 @@
+    """
+    return quote(string, safe="-_.~a-zA-Z0-9")
+
+
+def test_pubchem_server(verbosity):
+    """
+    This function is used to test the connection to the PubChem server.
+    """
+    response = requests.get(
+        "https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/1/cids/TXT", timeout=10
+    )
+    if response.status_code == 200 and response.text.strip() == "1":
+        if verbosity > 1:
+            print("PubChem Server is working fine.")
+    else:
+        raise ConnectionError(
+            "No connection could be established. Check if you have access to the internet."
+        )
+
+
+def search_compound(compound, input_format):
+    """
+    This function is used to search the compound in the PubChem database.
+    """
+    base_url = "https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/"
+    search_url = ""
+    if input_format == "inchi":
+        # Prepare the data payload
+        data = {"inchi": compound}
+        search_url = f"{base_url}inchi/{rawurlencode(compound)}/cids/TXT"
+    elif input_format == "name":
+        search_url = f"{base_url}name/{rawurlencode(compound)}/cids/TXT"
+    elif input_format == "cid":
+        search_url = f"{base_url}cid/{compound}/property/IUPACname/TXT"
+    elif input_format in ["smile", "smiles"]:
+        search_url = f"{base_url}smiles/{rawurlencode(compound)}/cids/TXT"
+    elif input_format in ["cas", "regid"]:
+        search_url = f"{base_url}xref/RegistryID/{rawurlencode(compound)}/cids/TXT"
+
+    if input_format != "inchi":
+        response = requests.get(search_url, timeout=10)
+    else:
+        response = requests.post(search_url, data=data, timeout=10)
+    return response.text.strip()
+
+
+### Class that combines the compound specific data
+class Compound:
+    """
+    This class handles everything related to a compound.
+    """
+
+    HLGAP_THR: float = 0.5
+
+    def __init__(self, name: str, cid: str, wdir: Path, xtb_path: Path, verbosity: int):
+        """
+        This function is used to initialize the compound object.
+        """
+        self.name = name
+        self.cid = cid
+        self.wdir = wdir
+        self.xtb_path: Path = xtb_path
+        self.verbosity = verbosity
+        self.struc: Path | None = None
+        self.chrg: int | None = None
+        self.hlgap: float | None = None
+
+    def __str__(self):
+        return f"{self.cid}\t{self.name}{"\t"+str(self.struc.resolve()) if self.struc else ''}"
+
+    def retrieve_3d_sdf(self):
+        """
+        This function is used to retrieve the 3D conformer data from the PubChem database.
+        """
+        response = requests.get(
+            f"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/cid/{self.cid}/sdf?record_type=3d",
+            timeout=10,
+        )
+        if "PUGREST.NotFound" in response.text:
+            try:
+                self.twodthreed()
+            except Exception as exc:
+                if self.verbosity > 1:
```

Comment:
An error message should be as verbose as possible imo. No need to hide it behind a verbosity level.

`raise Exception(f"Error in 2D to 3D structure conversion for CID {self.cid}.") from e`

Reply:
This block is try-except block is also somewhat redundant. You only raise one exception in `self.twodthreed()`. Why not only write the error message there?